cmd/broadcaster-cli/app/utxos/broadcast/broadcast.go

```diff
@@ -109,44 +100,32 @@ var Cmd = &cobra.Command{
 			return fmt.Errorf("failed to create client: %v", err)
 		}
 
-		fundingKeySets := make([]*keyset.KeySet, len(keyFiles))
-		for i, kf := range keyFiles {
-
-			fundingKeySet, _, err := helper.GetKeySetsKeyFile(kf)
-			if err != nil {
-				return fmt.Errorf("failed to get key sets: %v", err)
-			}
-
-			fundingKeySets[i] = fundingKeySet
-		}
-
 		wocClient := woc_client.New(woc_client.WithAuth(wocApiKey), woc_client.WithLogger(logger))
 
-		rateBroadcaster, err := broadcaster.NewMultiKeyRateBroadcaster(logger, client, fundingKeySets, wocClient, isTestnet,
-			broadcaster.WithFees(miningFeeSat),
-			broadcaster.WithCallback(callbackURL, callbackToken),
-			broadcaster.WithFullstatusUpdates(fullStatusUpdates),
-			broadcaster.WithBatchSize(batchSize),
-		)
+		rateBroadcaster, err := broadcaster.NewMultiKeyRateBroadcaster(logger, client, keySets, wocClient, isTestnet, broadcaster.WithFees(miningFeeSat), broadcaster.WithCallback(callbackURL, callbackToken), broadcaster.WithFullstatusUpdates(fullStatusUpdates), broadcaster.WithBatchSize(batchSize))
 		if err != nil {
 			return fmt.Errorf("failed to create rate broadcaster: %v", err)
 		}
 
-		go func() {
-			signalChan := make(chan os.Signal, 1)
-			signal.Notify(signalChan, os.Interrupt) // Signal from Ctrl+C
-			<-signalChan
-
-			logger.Info("Shutting down broadcaster")
-			rateBroadcaster.Shutdown()
-		}()
-
-		logger.Info("Starting broadcaster", slog.Int("rate [txs/s]", rateTxsPerSecond), slog.Int("batch size", batchSize))
 		err = rateBroadcaster.Start(rateTxsPerSecond, limit)
 		if err != nil {
 			return fmt.Errorf("failed to start rate broadcaster: %v", err)
 		}
 
+		shutdownChan := make(chan struct{}) // Channel to signal completion of shutdown
+		signalChan := make(chan os.Signal, 1)
+		signal.Notify(signalChan, os.Interrupt) // Listen for Ctrl+C
+
+		go func() {
+			<-signalChan // Wait for signal
+			fmt.Println("Shutdown signal received. Shutting down the rate broadcaster.")
+			rateBroadcaster.Shutdown() // Shutdown the broadcaster
+			close(shutdownChan)        // Signal that shutdown is complete
+		}()
+
+		// Block until shutdown is complete
+		<-shutdownChan
```

Comment:
If `rateBroadcaster.Start()` function completes without aborting with Ctrl+C, then `Shutdown()` on line 122 will not be called and this line 127 will block forever.

What is the aim of this change?

Reply:
When RateBroadcaster.Start() function completes ,the shutdownChan will never be closed because the goroutine waiting for the Ctrl+C signal will still be waiting indefintely. The aim is to shutdown when an interrupt signal Ctrl+C is received

Reply:
> The aim is to shutdown when an interrupt signal Ctrl+C is received

That was already given in the previous code 

```
		go func() {
			signalChan := make(chan os.Signal, 1)
			signal.Notify(signalChan, os.Interrupt) // Signal from Ctrl+C
			<-signalChan

			logger.Info("Shutting down broadcaster")
			rateBroadcaster.Shutdown()
		}()
```


I just tested it. With this change `broadcaster-cli utxos broadcast` only stops when we do a Ctrl+C. 
```
broadcaster-cli utxos broadcast --rate=2 --batchsize=5 --limit=10
Jul 22 11:41:58.209 INF stats txs=0 target=20 percentage=0 connections=0 utxos=20000
Jul 22 11:42:00.210 INF stats txs=5 target=20 percentage=25 connections=0 utxos=16735
Jul 22 11:42:02.212 INF stats txs=10 target=20 percentage=50 connections=0 utxos=13466
Jul 22 11:42:04.213 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:04.897 INF limit reached address=mpJGW95LADBSkBmYu8GkJDFkH2h8cdZ8Bd total=10 limit=10
Jul 22 11:42:04.898 INF shutting down broadcaster address=mpJGW95LADBSkBmYu8GkJDFkH2h8cdZ8Bd
Jul 22 11:42:05.708 INF limit reached address=muEb43678ZWcusGiEJ8JkgQg9TDTR88jef total=10 limit=10
Jul 22 11:42:05.709 INF shutting down broadcaster address=muEb43678ZWcusGiEJ8JkgQg9TDTR88jef
Jul 22 11:42:06.210 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:08.208 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:10.209 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:12.208 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:14.210 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:16.209 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
Jul 22 11:42:18.208 INF stats txs=20 target=20 percentage=100 connections=0 utxos=13464
^CShutdown signal received. Shutting down the rate broadcaster.
Broadcaster shutdown complete.

```

But we want it to stop if either
1. The function completes
2. Ctrl+C

Reply:
pushing a fix for this now

Reply:
<img width="699" alt="Screenshot 2024-07-22 at 11 19 12" src="https://github.com/user-attachments/assets/27b8d0c5-5a12-4e6e-b39f-a83c6fa7d014">